### PR TITLE
Online testing mode and statistics display

### PR DESCRIPTION
**Pull request by Arden488**
This PR contains two small PRs (I've combined them):
1. Show statistics screen with data from database
This change is based on Nathan's code for DB. We've spotted some problems there, but it is working and I've merged my branch with it.
2. Online test mode.
It is disabled by default. To enable it you need to edit a variable in settings.js.
Full path: `src/main/resources/com/toptrumps/online/scripts/config/settings.js`
Change 
`TEST_MODE: false `
to 
`TEST_MODE: true`
photo_2020-02-13 18 42 03

Don't forget to make hard reload of your browser (as this mode is based entirely on javascript and it was previously cached in your browser)
During test mode animations are fast and some of them are disabled entirely (no need to flip the card as they change very fast).
For your convenience, when the test mode is enabled - you can find useful info in the console (even if it wasn't open when the error occurred).
photo_2020-02-13 18 42 00

### CODE REVIEW

**Review by rjs68 — APPROVED** [review]
Awesome work! Thanks for the test mode - as much as I love the animations this is gonna be a life saver! 💯

**Review by hughwin — APPROVED** [review]
I spent 40 mins in the Libray testing it the other day. It's going to help so much.

File: src/main/java/com/toptrumps/db/FullStats.java [hunk 1]
```diff
@@ -0,0 +1,86 @@
...
+import com.toptrumps.core.player.Player;
+import com.toptrumps.online.api.request.PlayerState;
+
+import java.sql.ResultSet;
+import java.sql.SQLException;
+import java.util.HashMap;
+import java.util.Map;
+
+// TODO: Refactor this class!!!
+public class FullStats {
```
Comment (BelfDev) [inline src/main/java/com/toptrumps/db/FullStats.java:14]: We really gotta refactor this XD

File: src/main/java/com/toptrumps/db/PlayersList.java [hunk 2]
```diff
@@ -0,0 +1,26 @@
+package com.toptrumps.db;
+import java.util.*;
+import com.toptrumps.core.player.Player;
+public class PlayersList{
+
+    public PlayersList(){
```
Comment (BelfDev) [inline src/main/java/com/toptrumps/db/PlayersList.java:6]: We must get rid of this PlayerList class later on

File: src/main/resources/com/toptrumps/online/scripts/config/settings.js [hunk 3]
```diff
@@ -0,0 +1,15 @@
+const APP = {
```
Comment (BelfDev) [inline src/main/resources/com/toptrumps/online/scripts/config/settings.js:1]: Awesome!

File: src/main/resources/com/toptrumps/online/scripts/game.js [hunk 4]
```diff
@@ -26,6 +30,7 @@ const Game = (($) => {
         const numberOfOpponents = getNumberOfOpponents();
 
         const request = NetworkHelper.makeRequest("api/game", { numberOfOpponents });
+        Logger.output("Number of opponents", "init", numberOfOpponents);
```
Comment (BelfDev) [inline src/main/resources/com/toptrumps/online/scripts/game.js:33]: Hmm, we should be careful with this name. This Logger cannot be mistaken by the Logger class from server-side.

**Review by BelfDev — APPROVED** [review]
Excellent!

File: src/main/resources/com/toptrumps/online/scripts/game.js (hunk 4 above)
Comment (Arden488) [reply src/main/resources/com/toptrumps/online/scripts/game.js:33]: I can add a lot of comments in there. However, if you have doubts, I can get rid of it before we'll be ready to upload.

File: src/main/java/com/toptrumps/db/PlayersList.java (hunk 2 above)
Comment (nathanieldschneider) [reply src/main/java/com/toptrumps/db/PlayersList.java:6]: Already done that locally :)

**Pull request merged**